src/wallet/wallet2.cpp

```diff
@@ -1147,13 +1155,16 @@ void wallet2::process_new_transaction(const crypto::hash &txid, const cryptonote
     // additional tx pubkeys and derivations for multi-destination transfers involving one or more subaddresses
     std::vector<crypto::public_key> additional_tx_pub_keys = get_additional_tx_pub_keys_from_extra(tx);
     std::vector<crypto::key_derivation> additional_derivations;
-    for (size_t i = 0; i < additional_tx_pub_keys.size(); ++i)
+    if (pk_index == 1)
```

Comment:
I see now how that strange amount was caused (i.e. the tx had two identical tx pubkeys which falsely doubled the received amount), but I wonder if this part is necessary/relevant.

Reply:
Those keys only need to be generated/checked once, same idea, though these don't actually cause a problem technically so its mostly a speedup.

Reply:
Hmm, but I doubt there would be any real speedup, because the subaddress PR was merged much later (October 2017) than when the cold signing bug was found and fixed (December 2016); it wouldn't have been possible for anyone to create a tx which mistakenly has two tx public keys due to the cold signing bug and then to simultaneously attach additional tx public keys which are specific to the subaddresses scheme.

Not that I'm strongly against it, just feels a little strange to me.